src/main/java/com/chaosthedude/naturescompass/util/BiomeSearchWorker.java

```diff
@@ -86,19 +104,13 @@ public boolean doWork() {
 			if (biomeAtPosKey != null && biomeAtPosKey.equals(biomeKey)) {
 				finish(true);
 				return false;
+			} else if (biomeAtPosKey != lastBiomeKey) {
+				lastBiomeKey = biomeAtPosKey;
+				lastStep = 0;
```

Comment:
Not sure what the point of resetting `lastStep` is when you're unconditionally assigning it to `step` below.

Reply:
Woops, missed an else.